### ISSUE TITLE
[BUG] OOM when invoking normalize_characters on a relatively small dataframe

### ISSUE THREAD

**Issue by miguelusque** [rapidsai/cudf#10858]
**Describe the bug**
Hi, I am facing the following OOM when invoking normalize_characters() on a relatively small dataset, which uses less than 2 GB of VRAM on a 16GB v100.

In case it helps, normalize_spaces() works without issues in the same dataframe.

Please, find below how to reproduce it.
 
![image](https://user-images.githubusercontent.com/26169771/168416343-b712b4bf-e92b-44b4-babb-6f303130a87e.png)

**Steps/Code to reproduce bug**

> import cudf
> import pandas as pd
> 
> df = cudf.DataFrame({"text": pd.util.testing.rands_array(256, 5000000)})
> #df["text"] = df["text"].str.normalize_spaces()
> df["text"] = df["text"].str.normalize_characters(do_lower=False)
> 

**Expected behavior**
No OOM

**Environment overview (please complete the following information)**
 - Environment location: [Bare-metal, Docker, Cloud(specify cloud provider)]
 - Method of cuDF install: [conda, Docker, or from source]
   - If method of install is [Docker], provide `docker pull` & `docker run` commands used

**Environment details**
DGX-v100, cudf 22.04

**Comment by davidwendt** [comment]
Yes, this is the expected memory overhead. From what I can tell, the working memory size math looks like the following:
```
(bytes  * 3 * sizeof(uint32)) +  (bytes * sizeof(uint32)) = (bytes * 12) + (bytes * 4) = bytes * 16
where bytes = number of bytes in the chars child column
```
So the working memory required is ~16x the number of bytes in the strings column.
This does not include any working memory required by various thrust calls or the size of the output column.
My recommendation would be to slice the column and then call normalize on the slices. 
Use the above math to help determine the slices.

**Comment by miguelusque** [comment]
Hi! 

I think this issue is still relevant. IMHO, a 16x memory requirement for a normalize characters operation is a bit high.

Could anyone please consider an alternative implementation? Thanks!

**Comment by vyasr** [comment]
@davidwendt I see that you've assigned yourself, did you spend any time considering alternative implementations here? Is this an issue for which you think we could use memory more sparingly with a suitable alternative implementation, or is there some fundamental blocker that would make this unfixable?

**Comment by davidwendt** [comment]
Yes, I believe there are alternative implementations that could reduce memory usage.
There is no fundamental blocker -- just priority and time.

**Comment by vyasr** [comment]
Cool. Marking as backlogged.

**Comment by davidwendt** [comment]
This PR https://github.com/rapidsai/cudf/pull/17818 has reduced the memory requirement from 16x the input size to 12x.
Further reduction may be possible but would affect performance.
